### PR TITLE
CNS-872: IPRPC part 5 - Unit tests

### PR DESCRIPTION
**Pull request by oren-lava**

### CODE REVIEW

File: x/rewards/keeper/iprpc_test.go [hunk 1]
```diff
@@ -0,0 +1,642 @@
...
+	res, err = ts.QueryRewardsIprpcSpecReward("")
+	require.NoError(t, err)
+	require.ElementsMatch(t, afterProviderServiceExpectedResults, res.IprpcRewards)
+}
+
+// TestIprpcRewardObjectsUpdate tests that the IPRPC reward objects' management works as expected:
+// Scenarios:
+// 0. fund iprpc pool for 2 months, current should be 0 and first iprpc reward should be with id=1 (fund is always for the next month)
+// 1. there is no service to eligible subscriptions, month passes -> current shouldn't increment and there should be no IPRPC object
+// 2. provider provides service for eligible subscription, month passes -> current should increment by 1 and a new IPRPC reward should be created with id=current
```
Comment (omerlavanet) [inline x/rewards/keeper/iprpc_test.go:331]: let's add a scenario fund + serve and pay providers + fund more later + serve get more rewards from both funds

File: x/rewards/keeper/iprpc_test.go [hunk 2]
```diff
@@ -0,0 +1,642 @@
...
+			}
+		})
+	}
+}
+
+// TestIprpcEligibleSubscriptions tests that IPRPC CU is counted only if serviced an eligible subscription
+// Scenarios:
+// 0. assume two providers: p1, p2 and two consumers: c1, c2. Only c1 is IPRPC eligible
+// 1. p1 provides service for both consumers, p2 provides service for c1 -> IPRPC reward should divide equally between p1 and p2
+// 2. both providers provide service for c2 -> No IPRPC rewards should be given
```
Comment (omerlavanet) [inline x/rewards/keeper/iprpc_test.go:461]: let's add another month after the funding ends and see they are not getting more rewards after

**Review by omerlavanet — CHANGES_REQUESTED** [review]

**Comment by Yaroms** [comment]
OMER: please add a unitest that would find this by setting cost = mincost *1.1 and duration 2

File: x/rewards/keeper/iprpc_test.go (hunk 2 above)
Comment (oren-lava) [reply x/rewards/keeper/iprpc_test.go:470]: done. see https://github.com/lavanet/lava/commit/85ba6891d7cb69e895c6500e8115b27858e3a3cb

File: x/rewards/keeper/iprpc_test.go (hunk 1 above)
Comment (oren-lava) [reply x/rewards/keeper/iprpc_test.go:339]: done. see https://github.com/lavanet/lava/commit/d24e2b9b06ce7e85d048887903e734174a6a55b2

**Comment by oren-lava** [comment]
for the unit test asked in the comment, see here: https://github.com/lavanet/lava/commit/1a0e230004c9aa514fa8de5ebdd7b2bf5399c5d2

**Pull request merged**